metrics/ci_integrations/test/client/github_actions/github_actions_client_test.dart

```diff
@@ -309,6 +309,124 @@ void main() {
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() applies the default per page parameter if it is not specified",
+      () async {
+        final interactionResult = await client.fetchWorkflowRunsWithConclusion(
+          workflowId,
+        );
+        final runsPage = interactionResult.result;
+
+        expect(runsPage.perPage, isNotNull);
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() returns a workflow runs page with the given per page parameter",
+      () async {
+        const expectedPerPage = 3;
+
+        final interactionResult = await client.fetchWorkflowRunsWithConclusion(
+          workflowId,
+          perPage: expectedPerPage,
+        );
+        final runsPage = interactionResult.result;
+
+        expect(runsPage.perPage, equals(expectedPerPage));
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() fetches the first page if the given page parameter is null",
+      () async {
+        final interactionResult = await client.fetchWorkflowRunsWithConclusion(
+          workflowId,
+          page: null,
+        );
+        final runsPage = interactionResult.result;
+
+        expect(runsPage.page, equals(1));
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() fetches the first page if the given page parameter is less than zero",
+      () async {
+        final interactionResult = await client.fetchWorkflowRunsWithConclusion(
+          workflowId,
+          page: -1,
+        );
+        final runsPage = interactionResult.result;
+
+        expect(runsPage.page, equals(1));
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() returns a workflow runs page with the given page parameter",
+      () async {
+        const expectedPage = 2;
+
+        final interactionResult = await client.fetchWorkflowRunsWithConclusion(
+          workflowId,
+          page: expectedPage,
+        );
+        final runsPage = interactionResult.result;
+
+        expect(runsPage.page, equals(expectedPage));
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() returns a workflow runs page with the next page url",
+      () async {
+        final interactionResult = await client.fetchWorkflowRunsWithConclusion(
+          workflowId,
+        );
+        final runsPage = interactionResult.result;
+
+        expect(runsPage.nextPageUrl, isNotNull);
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() returns a workflow runs page with the null next page url if there are no more pages available",
+      () async {
+        final interactionResult = await client.fetchWorkflowRunsWithConclusion(
+          workflowId,
+          perPage: 100,
+        );
+        final runsPage = interactionResult.result;
+
+        expect(runsPage.nextPageUrl, isNull);
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() returns a workflow runs page with a list of workflow runs",
```

Comment:
Should we test that this method fetches the workflow runs with the given conclusion?